include/rocksdb/utilities/cache_dump_load.h

```diff
@@ -0,0 +1,77 @@
+//  Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+
+#pragma once
+
+#include <set>
+
+#include "rocksdb/cache.h"
+#include "rocksdb/env.h"
+#include "rocksdb/io_status.h"
+#include "rocksdb/secondary_cache.h"
+#include "rocksdb/table.h"
+
+namespace ROCKSDB_NAMESPACE {
+
+class CacheDumpWriter {
+ public:
+  virtual ~CacheDumpWriter() = default;
+
+  virtual IOStatus Write(const Slice& data) = 0;
+  virtual IOStatus Close() = 0;
+  virtual uint64_t GetFileSize() = 0;
+};
+
+class CacheDumpReader {
+ public:
+  virtual ~CacheDumpReader() = default;
+  virtual IOStatus Read(size_t len, std::string* data) = 0;
+  virtual size_t GetOffset() const = 0;
+  virtual IOStatus Close() = 0;
+};
+
+// CacheDumpOptions is used to control DumpCache()
+struct CacheDumpOptions {
+  // The env pointer
+  Env* env;
```

Comment:
Why is the env option needed here?

Reply:
It is mainly used to get the SystemClock() for the timestamp.

Reply:
Can it just take a pointer to SystemClock then? Using the Env here is a little confusing since the DB might be on one Env, while the cache dump may happen on another (most likely Posix) Env.

Reply:
@anand1976 Sure. I will change that. if so, we will only have SystemClock* clock as the variable member of CacheDumpOption?

Reply:
Right, and users can call ```env->GetSystemClock()``` to get the pointer.